### PR TITLE
OTC-881: Product code uniqueness query and check.

### PR DESCRIPTION
**Pull request by wzglinieckisoldevelo**
https://openimis.atlassian.net/browse/OTC-881

Changes:
- Added query for checking of the uniqueness of the product code
- Backend now blocks saving if code is not unique
- Query checks for user permission before responding

### CODE REVIEW

File: product/gql_mutations.py [hunk 1]
```diff
@@ -61,6 +62,14 @@ def create_or_update_product(user, data):
     deductibles = extract_deductibles(data)
     ceilings = extract_ceilings(data)
 
+    incoming_code = data.get('code')
+    current_product = Product.objects.filter(uuid=product_uuid)
```
Comment (malinowskikam) [inline product/gql_mutations.py:66]: You are missing .first() here

```suggestion
    current_product = Product.objects.filter(uuid=product_uuid).first()
```

File: product/services.py [hunk 2]
```diff
@@ -150,3 +150,10 @@ def set_product_services(product, services, user):
             audit_user_id=user.id_for_audit,
             **service,
         )
+
+
+def check_unique_code_product(code):
+    from .models import Product
```
Comment (malinowskikam) [inline product/services.py:156]: You can import that at the top of that's the same module

**Review by malinowskikam — CHANGES_REQUESTED** [review]
Please chekc out my comments

**Review by malinowskikam — APPROVED** [review]
Looks good to me

**Review by delcroip — APPROVED** [review]

**Pull request merged**